test_runner/regress/test_storage_controller.py

```diff
@@ -3093,6 +3093,46 @@ def reconfigure_node_again():
     wait_until(reconfigure_node_again)
 
 
+def test_ps_unavailable_after_delete(neon_env_builder: NeonEnvBuilder):
+    neon_env_builder.num_pageservers = 3
+
+    env = neon_env_builder.init_start()
+
+    def assert_nodes_count(n: int):
+        nodes = env.storage_controller.node_list()
+        assert len(nodes) == n
+
+    # Nodes count must remain the same before deletion
+    assert_nodes_count(3)
+
+    ps = env.pageservers[0]
+    env.storage_controller.node_delete(ps.id)
+
+    # After deletion, the node count must be reduced
+    assert_nodes_count(2)
+
+    ps.restart()
+
+    # After deleted pageserver restart, the node count must remain the same
+    assert_nodes_count(2)
+
+    tombstones = env.storage_controller.tombstone_list()
+    assert len(tombstones) == 1 and tombstones[0]["id"] == ps.id
+
+    env.storage_controller.tombstone_delete(ps.id)
+
+    tombstones = env.storage_controller.tombstone_list()
+    assert len(tombstones) == 0
+
+    # Pageserver have to restart to be able to register again
```

Comment:
This is actually not true. Pagservers attempt the re-attach call in a loop which blocks further start-up. I would remove the tombstone and `wait_until(assert_nodes_count(3))`